collect_app/src/androidTest/java/org/odk/collect/android/instrumented/database/helpers/InstancesDatabaseHelperTest.java

```diff
@@ -29,7 +31,7 @@
 @RunWith(Parameterized.class)
 @Ignore("`Parameterized` causes problems for Firebase sharding. Probably need to replace this at JUnit level")
 public class InstancesDatabaseHelperTest extends SqlLiteHelperTest {
```

Comment:
It looks like these tests all fail. Should we rework or delete (if we don't need them) as part of this?

Reply:
I think it would be good to have such tests that confirm that upgrading/downgrading databases work well so I would keep it.

Reply:
Totally agree that we want that coverage. With forms, it seems we have that covered in `FormDatabaseMigratorTest` so we could add a similar test for instances and remove this one (so replace the instrumented test with a Robolectric test). That said this PR definitely offers a bunch of good stuff, so we could just create an issue for that?

Reply:
Yeah I agree.